sdk/go/common/resource/plugin/plugin.go

```diff
@@ -404,15 +409,20 @@ func execPlugin(ctx *Context, bin, prefix string, kind apitype.PluginKind,
 		}, nil
 	}
 
+	if len(env) == 0 {
+		// If no environment is provided, use the current environment.
+		env = os.Environ()
+	}
+	env = append(env, "PULUMI_ROOT_DIRECTORY="+rootDir)
+	env = append(env, "PULUMI_PROGRAM_DIRECTORY="+pwd)
```

Comment:
This should match the ROOT/PROGRAM_DIRECTORY info from above. I shouldn't get different envvars just because I'm a binary vs run from source. The root/program directory should be the same, the folder where _I_ am. Not where the main pulumi program is.

Reply:
I thought the idea was that `PULUMI_ROOT_DIRECTORY` is where the `Pulumi.yaml` lives, and `PULUMI_PROGRAM_DIRECTORY` where the main program is, and they would be different when e.g. `main` is set in `Pulumi.yaml`.  

For executing programs we would otherwise always get `PULUMI_ROOT_DIRECTORY` to be the same as the path of the language host binary, which is usually `~/.pulumi/bin`.  That seems less useful than knowing where the pulumi program lives.  I think knowing where the program lives is also what is requested in https://github.com/pulumi/pulumi/issues/16843, and that's what I'm doing here.